crates/core_simd/src/eq.rs

```diff
@@ -71,3 +73,37 @@ macro_rules! impl_mask {
 }
 
 impl_mask! { i8, i16, i32, i64, isize }
+
+impl<T, const LANES: usize> SimdPartialEq for Simd<*const T, LANES>
```

Comment:
This seems okay. Are there any caveats, e.g. if the pointers get evaluated at const time, that we should document?

Reply:
I'm not sure it has any caveats beyond what's already true for `const fn` or whatever.  Did you have anything specific in mind?

Reply:
I am thinking of the pointer indeterminacy problem during CTFE. Things that would share addresses during runtime may not during compile-time function execution, and vice versa.

Reply:
Honestly, not sure what the implications would be other than since the SIMD intrinsics aren't const (yet) so we know any pointers for now will be runtime only.  If you do something bad with CTFE before putting it in a vector, that's probably out of scope for now.

Reply:
I think it's fair to assume all vector operations should mirror scalar operations in that regard, whatever the implications are.

Reply:
Fair enough. I think it's mostly something we should keep in mind while exploring the API surface.